OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -169,6 +171,8 @@ protected virtual void Tick(Actor self)
 			if (IsTraitDisabled)
 				return;
 
+			++ticksSinceLastShot;
```

Comment:
Style nit: Try to avoid incrementing an integer without bound because it could in principle overflow and become negative.  Thats not likely in practice to happen here, but it is still bad style.  You can put this behind an `if (ticksSinceLastShot < Weapon.ReloadDelay)`.